### PR TITLE
Removed Dictionary<K,V>.GetValueOrDefault  From ref contract

### LINKED ISSUE #17917: Remove Dictionary<K,V>.GetValueOrDefault instance methods and IDictionary<K,V>.GetValueOrDefault extension methods
`IDictionary<TKey,TValue>` already has:

```c#
partial class CollectionExtensions
{
  public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key);
  public static TValue GetValueOrDefault<TKey, TValue>(this IDictionary<TKey, TValue> dictionary, TKey key, TValue defaultValue);
}
```

As extension methods in `CollectionExtensions`. 

In this discussion: #17275 we've decided to remove this instance methods, see: https://github.com/dotnet/corefx/issues/17275#issuecomment-291636863

cc: @danmosemsft @ianhays

### PR DESCRIPTION
**Pull request by Pothulapati**
As Per the Issue  #17917 , GetValueOrDefault Api's from the ref contract in the Dictionary class has been removed.

Closes #17917

### CODE REVIEW

File: src/System.Collections/src/ApiCompatBaseline.txt [hunk 1]
```diff
@@ -0,0 +1 @@
+Total Issues: 0
```
Comment (stephentoub) [inline src/System.Collections/src/ApiCompatBaseline.txt]: Is this file needed?

File: src/System.Collections/ref/System.Collections.cs [hunk 2]
```diff
@@ -107,8 +107,6 @@ public partial class Dictionary<TKey, TValue> : System.Collections.Generic.IColl
         void System.Collections.IDictionary.Remove(object key) { }
         System.Collections.IEnumerator System.Collections.IEnumerable.GetEnumerator() { throw null; }
         public bool TryGetValue(TKey key, out TValue value) { throw null; }
-        public TValue GetValueOrDefault(TKey key) { throw null; }
-        public TValue GetValueOrDefault(TKey key, TValue defaultValue) { throw null; }
```
Comment (stephentoub) [inline src/System.Collections/ref/System.Collections.cs]: Are there any corresponding tests to be deleted?

Comment (safern) [inline src/System.Collections/ref/System.Collections.cs]: GetValueOrDefault is an extension Method for IDictionary that is why we are removing it so the tests would still pass. I'm verifying desktop tests though.

File: src/System.Collections/src/ApiCompatBaseline.txt (hunk 1 above)
Comment (safern) [inline src/System.Collections/src/ApiCompatBaseline.txt]: +1. We can delete this file

File: src/System.Collections/ref/System.Collections.cs (hunk 2 above)
Comment (safern) [inline src/System.Collections/ref/System.Collections.cs]: This tests are only included in netcoreapp so Desktop is fine as well.

Comment (stephentoub) [inline src/System.Collections/ref/System.Collections.cs]: > GetValueOrDefault is an extension Method for IDictionary that is why we are removing the instance methods, so the tests would still pass

Right, but I'm wondering if we now have duplicate tests, i.e. did we previously have tests before for Dictionary.GetValueOrDefault and for the IDictionary extension, and now they're duplicative and just testing the latter, such that we could remove half of them?

Comment (Pothulapati) [inline src/System.Collections/ref/System.Collections.cs]: I didn't find any tests for Dictionary.GetValueOrDefault. If you find any would be glad to remove them.

Comment (stephentoub) [inline src/System.Collections/ref/System.Collections.cs]: Ok. Thanks.

**Comment by Pothulapati** [comment]
@safern  Deleted the `ApiCompatBaseline.txt` as said but I couldn't find any tests regarding `IDictionary.GetValueOrDefault ` to remove but there were tests regarding `CollectionExtensions.GetValueOrDefault` .

keeping this Pull Request aside , Do I need to wait until this PR is merged to Remove the `IDictinary.GetValueOrDefault` Api's in both  `CoreRT`,`Coreclr ` and submit a PR?

File: src/System.Collections/ref/System.Collections.cs (hunk 2 above)
Comment (safern) [inline src/System.Collections/ref/System.Collections.cs]: From a quick scan I didn't find any either.

**Comment by safern** [comment]
@Pothulapati thanks for the update. I didn't find any tests to remove either. So one more thing to do, could you please fix your merge conflicts in `src/System.Runtime/src/ApiCompatBaseline.uapaot.txt` and `src/shims/ApiCompatBaseline.netcoreapp.netfx461.txt`?

One way to do this would be to discard your changes in this files, then rebase over corefx/master and finally do a fool `build.cmd` from your repo root and update the PR.

> Do I need to wait until this PR is merged to Remove the IDictinary.GetValueOrDefault Api's in both CoreRT,Coreclr and submit a PR?

You can start working on that and get your PRs ready (even submit them) but we will not be able to merge them until we merge this one (which is ready to merge once you fix the merge conflicts).

**Comment by safern** [comment]
@Pothulapati I think your last commit didn't fix the merge conflicts.

**Comment by Pothulapati** [comment]
@safern Take a look at the Code and I think its good to go

**Comment by safern** [comment]
Thanks @Pothulapati for your contribution. Now you should do steps 2 and 3 in https://github.com/dotnet/corefx/issues/17917#issuecomment-291749382
